packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx

```diff
@@ -64,8 +59,15 @@ function GridFilterInputDate(props: GridFilterInputDateProps) {
         shrink: true,
       }}
       inputRef={focusElementRef}
+      InputProps={{
+        ...(applying ? { endAdornment: <GridLoadIcon /> } : {}),
+        inputProps: {
+          max: type === 'datetime-local' ? '9999-12-31T23:59' : '9999-12-31',
+          ...inputProps,
```

Comment:
The top-level `inputProps` is still being applied inside `InputProps`. The following diff should fix this

```diff
diff --git a/packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx b/packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx
index 7e061a7ec..e8b4f1006 100644
--- a/packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx
+++ b/packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx
@@ -11,7 +11,7 @@ export type GridFilterInputDateProps = GridFilterInputValueProps &
 export const SUBMIT_FILTER_DATE_STROKE_TIME = 500;
 
 function GridFilterInputDate(props: GridFilterInputDateProps) {
-  const { item, applyValue, type, apiRef, focusElementRef, inputProps, InputProps, ...other } =
+  const { item, applyValue, type, apiRef, focusElementRef, InputProps, ...other } =
     props;
 
   const filterTimeout = React.useRef<any>();
@@ -61,11 +61,11 @@ function GridFilterInputDate(props: GridFilterInputDateProps) {
       inputRef={focusElementRef}
       InputProps={{
         ...(applying ? { endAdornment: <GridLoadIcon /> } : {}),
+        ...InputProps,
         inputProps: {
           max: type === 'datetime-local' ? '9999-12-31T23:59' : '9999-12-31',
-          ...inputProps,
+          ...InputProps?.inputProps,
         },
-        ...InputProps,
       }}
       {...other}
     />
```